cleanlab/filter.py

```diff
@@ -398,6 +423,165 @@ class 0, 1, ..., K-1.
+      if ``num_to_remove_per_class=k``, then either k-1, k, or k+1
+      examples may be removed for any class due to rounding error. If you need
+      exactly 'k' examples removed from every class, you should use
+      ``filter_by='prune_by_class'``.
+
+    min_examples_per_class : int, default=1
+      Minimum number of examples per class to avoid flagging as label issues.
+      This is useful to avoid deleting too much data from one class
+      when pruning noisy examples in datasets with rare classes.
+
+    confident_joint : np.ndarray, optional
+      An array of shape ``(num_classes,K, K)`` representing the confident joint, the matrix used for identifying label issues, which
+      estimates a confident subset of the joint distribution of the noisy and true labels, ``P_{noisy label, true label}``.
+      Entry ``(j, k)`` in the matrix is the number of examples confidently counted into the pair of ``(noisy label=j, true label=k)`` classes.
+      The `confident_joint` can be computed using :py:func:`count.compute_confident_joint <cleanlab.count.compute_confident_joint>`.
+      If not provided, it is computed from the given (noisy) `labels` and `pred_probs`.
+
+    n_jobs : optional
+      Number of processing threads used by multiprocessing. Default ``None``
+      sets to the number of cores on your CPU.
+      Set this to 1 to *disable* parallel processing (if its causing issues).
+      Windows users may see a speed-up with ``n_jobs=1``.
+
+    verbose : optional
+      If ``True``, prints when multiprocessing happens.
+
+    Returns
+    -------
+    label_issues : np.ndarray
+      If `return_indices_ranked_by` left unspecified, returns a boolean **mask** for the entire dataset
+      where ``True`` represents a label issue and ``False`` represents an example that is
+      accurately labeled with high confidence.
+      If `return_indices_ranked_by` is specified, returns a shorter array of **indices** of examples identified to have
+      label issues (i.e. those indices where the mask would be ``True``), sorting by likelihood that the corresponding label is correct is not supported yet.
+
+      Note
+      ----
+      Obtain the *indices* of label issues in your dataset by setting
+      `return_indices_ranked_by`.
+
+    """
+    num_classes = pred_probs.shape[1]
+    y_one = np.zeros((len(labels), num_classes)).astype(np.int32)
+    for class_num in range(0, len(labels)):
+        for j in range(0, len(labels[class_num])):
+            y_one[class_num][labels[class_num][j]] = 1
+    if return_indices_ranked_by is None:
+        bissues = np.zeros(y_one.shape).astype(bool)
+    else:
+        label_issues_list = []
+    for class_num in range(0, num_classes):
+        pred_probabilitites = np.stack([1 - pred_probs[:, class_num], pred_probs[:, class_num]]).T
+        if confident_joint is None:
+            conf = None
+        else:
+            if len(confident_joint.shape()) != 3:
```

Comment:
Can we make this condition more specific?
I think we should check that confident_joint is either (num_classes,k,k) or (k,k) and otherwise throw ValueError that it is not of an acceptable shape.

Reply:
I'm fine with this condition for now, but I agree that we should validate the tuple elements if we're going to throw ValueErrors.

Nevertheless, the shape property is not callable:
```suggestion
            if len(confident_joint.shape) != 3:
```